environment-dev.yml

```diff
@@ -5,6 +5,11 @@ dependencies:
   - pandas==1.5.2
   - numpy>=1.18.5
   - pyarrow>=4.0.1
+  - ray-default>=1.4.0,<2.1.0
+  # This will allow us to install ray==2.0.1 instead of ray==1.12.1;
```

Comment:
A little more detail on why limiting protobuf version allows us to install a later version of ray-default:
- ray-default of 1.12.1 and 2.1.0 versions hasn't upper bound for protobuf.
- ray-default of 1.13.0 and 2.0.1 versions has "<4.0.0" upper bound for protobuf.

Reply:
I suggest, if possible, not to interfere with the installation using conda and pip, since there may be situations when there is a difference in the distributed files of the same version, for example: https://github.com/conda-forge/ray-packages-feedstock/pull/78#issuecomment-1312572638.

Reply:
So we will remove protobuf restriction along with allowing  ray 2.1.0?

Reply:
That's the plan.

Or I can remove this restriction and we will temporarily use not the newest ray.

Reply:
we don't use it already (>=1.4.0,<2.1.0) or I miss something? :)

Reply:
I mean restriction for protobuf. 

Due to the fact that conda gives priority to the `protobuff` and ray does not support the newest versions of the `protobuff`, a newest `protobuff` and an old `ray` are installed, since `ray` has no restrictions on the `protobuff` (for its own internal reasons).

If we want to remove the restriction on ray right now, then we need to add a fix from https://github.com/modin-project/modin/pull/5283. The fix will include some restrictions for `grpcio` (it's a dependency of Ray).
Do not mind doing it right now, but it seemed to me that these are tasks that are better logically divided into different pull requests.